ddtrace/tracer.py

```diff
@@ -196,23 +197,31 @@ def global_excepthook(self, tp, value, traceback):
         """The global tracer except hook."""
 
     def get_call_context(self, *args, **kwargs):
-        # type: (...) -> Context
-        """
-        Return the current active ``Context`` for this traced execution. This method is
-        automatically called in the ``tracer.trace()``, but it can be used in the application
-        code during manual instrumentation like::
+        # type (...) -> Context
+        """Return the active ``Context`` for the current execution."""
+        active = self.context_provider.active(*args, **kwargs)
+        if isinstance(active, Context):
+            return active
+        elif isinstance(active, Span):
+            return active.context
+        else:
+            return Context()
```

Comment:
Sorry, I'm stupid. :)
Isn't that a problem to return a brand new one each time? I don't know why `get_call_context` would be use, but I can imagine somebody would call this twice to do two operations thinking they would do it on the same object and then it's not… or I'm wrong?

Reply:
For context, the reason we return a new context here is just to maintain the API - otherwise I'd prefer to return `None`. That said, you're right that returning a new context every time is problematic for something like the following

```python
ctx = tracer.get_call_context()
ctx.sampling_priority = USER_KEEP
...
with tracer.trace(...): ...

with tracer.trace(...): ...
```

Before this PR: the first trace will get the sampling priority applied but the second won't because the context is cleared after a trace finishes. (Even this seems unintuitive because the context is outside the scope of the first trace)

With this PR: neither of the traces will get the sampling priority applied which also seems wrong. This is why I'd prefer to return `None` from `get_call_context` so it's clear that there isn't a context.

So what this makes me think is that a breaking change should be make so that `get_call_context` can return `None`.

Reply:
> Even this seems unintuitive because the context is outside the scope of the first trace

You bet it is 😢 

I do agree with this breaking change TBH. I don't think there's much value in maintaining this unless you think there are 99% of the user using the API this way.